### PR TITLE
:sparkles: added reconciles_total metric

### PR DESCRIPTION
**Pull request by droot**
This PR adds a new metric 'reconciles_total' to track total reconciles per controller. 

Note: yet to add tests for it, sending in for early feedback.

### CODE REVIEW

**Comment by droot** [comment]
putting this PR on hold because want to fix the tests.

File: pkg/internal/controller/metrics/metrics.go [hunk 1]
```diff
@@ -47,6 +56,7 @@ var (
 func init() {
 	metrics.Registry.MustRegister(
 		QueueLength,
+		ReconcileTotal,
 		ReconcileErrors,
```
[the remark this answers is not in the document]
Comment (droot) [reply pkg/internal/controller/metrics/metrics.go]: Yes, they will be identical. Will do a separate PR where we drop support for ReconcileErrors.

**Comment by droot** [comment]
Added separate tests for `reconcile_total`. So this is good to go from my end. @DirectXMan12 @mengqiy PTAL.

File: pkg/internal/controller/metrics/metrics.go [hunk 2]
```diff
@@ -29,6 +29,15 @@ var (
 		Help: "Length of reconcile queue per controller",
 	}, []string{"controller"})
 
+	// ReconcileTotal is a prometheus counter metrics which holds the total
+	// number of reconciles per controller. It has two labels. controller label refers
+	// to the controller name and result label refers to the reconcile result i.e
+	// success, error, requeue, requeue_after.
+	ReconcileTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "controller_runtime_reconcile_total",
+		Help: "Total number of reconciles per controller",
```
Comment (mengqiy) [inline pkg/internal/controller/metrics/metrics.go]: nit: reconcile is a verb, i.e. s/reconciles/reconciliations

Comment (droot) [reply pkg/internal/controller/metrics/metrics.go]: Addressed.

**Comment by droot** [comment]
@mengqiy updated the PR, pl. take a look.

**Comment by mengqiy** [comment]
/lgtm
Linter is complaining a misspelling. Please fix.

**Comment by mengqiy** [comment]
@droot It seems goimports is not happy

**Comment by droot** [comment]
@mengqiy updated the tests and took care of the metalinter warnings. PTAL

**Pull request merged**